ibis/backends/pandas/execution/window.py

```diff
@@ -42,17 +42,19 @@ def _post_process_empty(
     timecontext: Optional[TimeContext],
 ) -> pd.Series:
     # This is the post process of the no groupby nor orderby window
-    # `result` could be a Series or a scalar. generated by `agg` method
-    # of class `Window`. For window without grouby or orderby, `agg`
-    # calls pands method directly. So if timecontext is present, we
-    # need to insert 'time' column into index for trimming the result.
-    # For cases when grouby or orderby is present, `agg` calls
+    # `result` could be a Series, DataFrame, tuple, or a scalar. generated
+    # by `agg` method of class `Window`. For window without grouby or
+    # orderby, `agg` calls pands method directly. So if timecontext is
+    # present, we need to insert 'time' column into index for trimming the
+    # result. For cases when grouby or orderby is present, `agg` calls
     # Ibis method `window_agg_built_in` and `window_agg_udf`, time
     # context is already inserted there.
     assert not order_by and not group_by
-    if isinstance(result, pd.Series):
+    if isinstance(result, (pd.Series, pd.DataFrame, tuple)):
```

Comment:
why don't we just test is_scalar? (and rever the else/if), and handle the list-like of scalar case